src/rules/quotemarkRule.ts

```diff
@@ -67,13 +71,15 @@ class QuoteWalker extends Lint.RuleWalker {
             const firstCharacter = text.charAt(0);
             const lastCharacter = text.charAt(text.length - 1);
 
-            if (this.quoteMark === QuoteMark.SINGLE_QUOTES) {
-                if (firstCharacter !== "'" || lastCharacter !== "'") {
-                    failure = this.createFailure(position, width, Rule.SINGLE_QUOTE_FAILURE);
-                }
-            } else if (this.quoteMark === QuoteMark.DOUBLE_QUOTES) {
-                if (firstCharacter !== "\"" || lastCharacter !== "\"") {
-                    failure = this.createFailure(position, width, Rule.DOUBLE_QUOTE_FAILURE);
+            const expectedQuoteMarkCharacter = getQuoteMarkCharacter(this.quoteMark);
```

Comment:
I fixed it.
https://github.com/palantir/tslint/commit/fab490f8fd79bd3693ceaac285d6c58a7beb3b1d#diff-ff9987f3d7e6a68c5bfdc281c84cbd31R72